example/src/demos/MagicMirror.tsx

```diff
@@ -0,0 +1,121 @@
+import * as THREE from 'three'
+import React, { useRef, useEffect, Suspense, useState } from 'react'
+import { Canvas, useFrame, useThree, createPortal } from '@react-three/fiber'
+import { OrbitControls, useGLTF, useFBO, Environment } from '@react-three/drei'
+import { useCallback } from 'react'
+
+function MagicMirror({ children, clearColor = 'white', ...props }: any) {
+  const ref = useRef<THREE.Mesh>(null!)
+  const { events } = useThree()
+  const [camera] = useState(() => new THREE.PerspectiveCamera(50, 1, 0.1, 1000))
+  const [scene] = useState(() => new THREE.Scene())
+  const fbo = useFBO()
+
+  camera.aspect = 4 / 5
+  camera.updateProjectionMatrix()
```

Comment:
Is it ok to have this in the body?

Reply:
its fine here. but i will remove the code later on, these things will become drei components.